bodo/pandas/base.py

```diff
@@ -44,8 +47,29 @@ def read_parquet(
             )
         else:
             assert False and "Unsupported option to read_parquet"
-    pr = plan_optimizer.LogicalGetParquetRead(path.encode())
-    return plan_optimizer.wrap_plan(get_pandas_schema(path), pr)
+
+    # Read Parquet schema and row count
+    # TODO: Make this more robust (e.g. handle Index, etc.)
+    use_hive = True
+    pq_dataset = get_parquet_dataset(
+        path,
+        get_row_counts=True,
+        storage_options=storage_options,
+        read_categories=True,
+        partitioning="hive" if use_hive else None,
+    )
+    arrow_schema = pq_dataset.schema
+    nrows = pq_dataset._bodo_total_rows
+
+    empty_df = pa.Table.from_pydict(
+        {k: [] for k in arrow_schema.names}, schema=arrow_schema
+    ).to_pandas()
+    empty_df.index = pd.RangeIndex(0)
+
+    metadata = LazyMetadata("dummy", empty_df, nrows, None)
+    return BodoDataFrame.from_lazy_metadata(
```

Comment:
Please use plan_optimizer.wrap_plan.  Let's try to centralize LazyMetadata or whatever it is we end up using rather than requiring this stuff in every location.  That was the goal of wrap_plan to have one single spot for the integration with existing lazy data structures.

Reply:
I agree. But the wrap_plan code is not working and making these parts robust is part of a follow up task.

Reply:
Why isn't it working?  Let's just get it working rather than doing things we have to fix later.

Reply:
The whole code will be replaced in the next immediate issue (too large to add to scope of this PR).